### PR TITLE
Weekend Challenge - Airport Challenge

### PR DESCRIPTION
**Pull request by phump81**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require_relative 'plane'
+
+class Airport
```
Comment (harrylb14) [inline lib/airport.rb:3]: Each method seems to be doing one job, lovely stuff

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  DEFAULT_CAPACITY = 50
+
+  attr_reader :capacity, :hangar, :weather
+
+  def initialize(capacity = DEFAULT_CAPACITY)
+    @hangar = []
+    @capacity = capacity
+    @weather = rand(10)
```
Comment (harrylb14) [inline lib/airport.rb:12]: potentially would have had Weather as its own class, as its a seperate concept to an Airport

File: lib/plane.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require_relative 'airport'
+
+class Plane
```
Comment (harrylb14) [inline lib/plane.rb:3]: Plane not doing anything, potentially needs some methods for it to have some use? for example, is it landed to make sure it can't land again

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+require 'airport'
```
Comment (harrylb14) [inline spec/airport_spec.rb:1]: tests are nice and concise, but potentially need more of them to cover every line of code (edge cases etc). for example can you check if the plane is in a specific airport, can you make sure it cant take off(/land) if its already taken off(/landed), does taking off remove the plane from the airport etc

File: lib/airport.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    @hangar << plane
+  end
+
+  def take_off
+    raise "Hangar is empty" if empty?
+
+    raise "May not take off due to inclement weather" if storm?
+
+    @hangar.pop
```
Comment (harrylb14) [inline lib/airport.rb:28]: Might want to review this method to take off a specific plane, rather than the last plane that was landed

File: README.md [hunk 6]
```diff
@@ -1,6 +1,14 @@
 Airport Challenge
```
Comment (harrylb14) [inline README.md:1]: excellent use of personalising the readme, definitely something I should have done!

File: spec/airport_spec.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  it 'changes the capacity' do
+    airport = Airport.new(70)
+    expect(airport.instance_variable_get(:@capacity)).to eq(70)
+  end
+
+  it 'prevents take off in stormy weather' do
+    allow(subject).to receive(:storm?) { false }
+    subject.land(Plane.new)
```
Comment (harrylb14) [inline spec/airport_spec.rb:29]: consider using doubles so there isnt Plane.new in airport spec

File: lib/airport.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "Plane has departed"
+  end
+
+  def full?
+    @hangar.count >= @capacity
+  end
+
+  def empty?
+    @hangar.count.zero?
```
Comment (harrylb14) [inline lib/airport.rb:37]: 1 line methods = lovely to see
